packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -867,7 +867,8 @@ function performConcurrentWorkOnRoot(root, didTimeout) {
   // Flush any pending passive effects before deciding which lanes to work on,
   // in case they schedule additional work.
   const originalCallbackNode = root.callbackNode;
-  const didFlushPassiveEffects = flushPassiveEffects();
+  // workInProgressTransitions should be what it in the previous render
+  const didFlushPassiveEffects = flushPassiveEffects(workInProgressTransitions);
```

Comment:
Instead of doing this make it a work loop variable (`pendingPassiveEffectsRemainingLanes`)